src/helpers.c

```diff
@@ -64,6 +64,34 @@ char *clearblank(size_t *vlen, char *string) {
   return buffer;
 }
 
+// Remove '[' and ']' from beginning and end
+char *stripbrackets(char *string) {
+  char *ptr = string;
+  size_t length = strlen(string) - 1;
+  if (*string == '[' && string[length] == ']') {
+    char *buffer = malloc(length);
+    ptr = buffer;
+    while (*(++string) != ']') { *buffer++ = *string; }
+    *buffer = 0;
+  }
+  return ptr;
+}
+
+// Add '[' and ']' to the given string
+char *addbrackets(char *string) {
+  char *ptr = string;
+  size_t length = strlen(string);
+  if (!(*string == '[' && string[length - 1] == ']')) {
+  char *buffer = malloc(length + 3);
```

Comment:
Wrong indentation.

Reply:
The function will either return a pointer to newly allocated memory or the existing one, so freeing the result may result in an invalid free.